### ISSUE TITLE
Streamline collection operations/actions

### ISSUE THREAD

**Issue by matthewcheok** [siawyoung/remerge#13]
Currently we write the following:

``` js
{
  models: {
    _: sampleModels,
  },
  'models[modelId]': {
    fields: {
      add: arrayInsertReducer,
    },
  },
}
```

Does the following make any sense at all?

``` js
{
  models: {
    _: sampleModels,
    '[modelId]': {
      fields: {
        add: arrayInsertReducer,
      },
    },
  },
}
```

I think it makes it clear that the actions happen on the elements of a collection (because its nested one level in.)

**Comment by matthewcheok** [comment]
Maybe this looks even better?

``` js
{
  models: {
    _: sampleModels,
    $modelId: {
      fields: {
        add: arrayInsertReducer,
      },
    },
  },
}
```

**Comment by siawyoung** [comment]
We can look for the accessor key in the action and drop the `[]` collection indicator completely:

```
const action = {
  type: "users.fields.add",
  userId: 0,
  data: { fieldName: ... }
}
```

**Comment by siawyoung** [comment]
Fixed in #14
